R/download_checkpoint.R

```diff
@@ -79,48 +203,84 @@ download_BERT_checkpoint <- function(model = c("bert_base_uncased",
                                     "bert_large_uncased",
                                     "bert_large_cased",
                                     "bert_large_uncased_wwm",
                                     "bert_large_cased_wwm",
                                     "bert_base_multilingual_cased",
                                     "bert_base_chinese")) {
-  checkpoint_url_map <- c(
-    "bert_base_uncased" = "https://storage.googleapis.com/bert_models/2018_10_18/uncased_L-12_H-768_A-12.zip",
-    "bert_base_cased" = "https://storage.googleapis.com/bert_models/2018_10_18/cased_L-12_H-768_A-12.zip",
-    "bert_large_uncased" = "https://storage.googleapis.com/bert_models/2018_10_18/uncased_L-24_H-1024_A-16.zip",
-    "bert_large_cased" = "https://storage.googleapis.com/bert_models/2018_10_18/cased_L-24_H-1024_A-16.zip",
-    "bert_large_uncased_wwm" = "https://storage.googleapis.com/bert_models/2019_05_30/wwm_uncased_L-24_H-1024_A-16.zip",
-    "bert_large_cased_wwm" = "https://storage.googleapis.com/bert_models/2019_05_30/wwm_cased_L-24_H-1024_A-16.zip",
-    "bert_base_multilingual_cased" = "https://storage.googleapis.com/bert_models/2018_11_23/multi_cased_L-12_H-768_A-12.zip",
-    "bert_base_chinese" = "https://storage.googleapis.com/bert_models/2018_11_03/chinese_L-12_H-768_A-12.zip"
+  root_url <- "https://storage.googleapis.com/bert_models/"
+  checkpoint_zips <- c(
+    "bert_base_uncased" = "2018_10_18/uncased_L-12_H-768_A-12.zip",
+    "bert_base_cased" = "2018_10_18/cased_L-12_H-768_A-12.zip",
+    "bert_large_uncased" = "2018_10_18/uncased_L-24_H-1024_A-16.zip",
+    "bert_large_cased" = "2018_10_18/cased_L-24_H-1024_A-16.zip",
+    "bert_large_uncased_wwm" = "2019_05_30/wwm_uncased_L-24_H-1024_A-16.zip",
+    "bert_large_cased_wwm" = "2019_05_30/wwm_cased_L-24_H-1024_A-16.zip",
+    "bert_base_multilingual_cased" =
+      "2018_11_23/multi_cased_L-12_H-768_A-12.zip",
+    "bert_base_chinese" = "2018_11_03/chinese_L-12_H-768_A-12.zip"
   )
 
+  checkpoint_url_map <- paste0(
+    root_url,
+    checkpoint_zips
+  )
+  names(checkpoint_url_map) <- names(checkpoint_zips)
+
   model <- match.arg(model)
   return(checkpoint_url_map[[model]])
 }
 
+
+# set_BERT_dir ------------------------------------------------------------
+
+#' Set the directory for BERT checkpoints
+#'
+#' Set a given `dir` as the default BERT checkpoint directory for this session,
+#' and create it if it does not exist.
+#'
+#' @inheritParams download_BERT_checkpoint
+#'
+#' @return A list with the previous value of `BERT.dir` (invisibly).
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'   set_BERT_dir("fake_dir")
+#' }
+set_BERT_dir <- function(dir) {
+  if (!file.exists(dir)) {
+    dir.create(dir) # nocov
+  }
+  dir <- normalizePath(dir)
+  options(BERT.dir = dir)
+}
+
+
+# Copied from `rlang` to avoid importing that package. Roxygen doesn't like it
+# and I'm not sure how to fix that, so install I'm not documenting.
```

Comment:
The "install" seems to be out of place here, unless I'm totally parsing this wrong.

Reply:
"install" still here?